File: javascript_code/line/algo/linkedlist/ListNode.js

class ListNode {
    constructor(val, next) {
        this.val = val
        this.next = next
    }

    toString() {
        let res = ""
        let curr = this
        while (curr != null) {
            res += `${curr.val} -> `
            curr = curr.next
        }
        res += "null"
        return res
    }
}

const fromArray = function (arr) {
    if (arr == null || arr.length == 0) return null
    const head = new ListNode(arr[0])
    let curr = head
    for (let i = 1; i < arr.length; i++) {
        curr.next = new ListNode(arr[i])
        curr = curr.next
    }
    return head
}

// 测试
//head = fromArray([23, 12, 11, 34])
//console.log(head.toString())

File: python_code/line/algo/linkedlist/ListNode.py

class ListNode:
    def __init__(self, val, next=None):
        self.val = val
        self.next = next

    def toString(self):
        res = ""
        curr = self
        while curr:
            res += str(curr.val) + "->"
            curr = curr.next
        res += "null"
        return res


def fromArray(arr):
    if arr is None or len(arr) == 0:
        return None
    head = ListNode(val=arr[0])
    curr = head
    for i in range(1, len(arr)):
        curr.next = ListNode(val=arr[i])
        curr = curr.next
    return head


if __name__ == '__main__':
    head = fromArray([23, 12, 11, 34])
    print(head.toString())